src/biogeophys/SoilFluxesMod.F90

```diff
@@ -224,59 +204,50 @@ subroutine SoilFluxes (bounds, num_urbanl, filter_urbanl, &
          endif
       end do
 
-      ! Set the column-average qflx_evap_soi as the weighted average over all patches
-      ! but only count the patches that are evaporating
-
-      do fc = 1,num_nolakec
-         c = filter_nolakec(fc)
-         topsoil_evap_tot(c) = 0._r8
-         sumwt(c) = 0._r8
-      end do
-
-      do pi = 1,max_patch_per_col
-         do fc = 1,num_nolakec
-            c = filter_nolakec(fc)
-            if ( pi <= col%npatches(c) ) then
-               p = col%patchi(c) + pi - 1
-               if (patch%active(p)) then
-                  topsoil_evap_tot(c) = topsoil_evap_tot(c) + qflx_evap_soi(p) * patch%wtcol(p)
-               end if
-            end if
-         end do
-      end do
+      ! evaporation from snow may be larger than available moisture
+      ! after flux update from tinc*cgrnd, repeat adjustment of ev_snow
+      do fp = 1,num_nolakep
+         p = filter_nolakep(fp)
+         c = patch%column(p)
+         j = col%snl(c)+1
+         ! snow layers
+         if (j < 1) then
+            ! assumes for j < 1 that frac_sno_eff > 0
+            evaporation_limit = (h2osoi_ice(c,j)+h2osoi_liq(c,j))/frac_sno_eff(c)
+            if (qflx_ev_snow(p)*dtime > evaporation_limit) then
+               ev_unconstrained = qflx_ev_snow(p)
+               qflx_ev_snow(p) = evaporation_limit/dtime
```

Comment:
I am thinking about it this way: each patch is spatially distinct wrt the atmosphere and sees the same amount of snow on its patch.  It should be allowed to remove all the snow on its patch, but not allowed to remove snow from the adjacent patches.  It's kind of arbitrary either way, but this way one doesn't need to include patch weights in the calculation.

Reply:
We didn't discuss this one. I'm fine with your rationale, but I just think it warrants a comment calling attention to the fact that we're being stricter in the limiting than we really need to be. (I'm thinking that this limiting is tricky enough already, so when we go back to it a year from now, it will help us understand it if all of these thoughts are recorded in comments.)

Reply:
Thanks for the comment!